pandas/core/base.py

```diff
@@ -831,7 +831,9 @@ def _map_values(self, mapper, na_action=None):
                 # If a dictionary subclass defines a default value method,
                 # convert mapper to a lookup function (GH #15999).
                 dict_with_default = mapper
-                mapper = lambda x: dict_with_default[x]
+                mapper = lambda x: dict_with_default[
+                    np.nan if type(x) == float and np.isnan(x) else x
```

Comment:
I think just checking `np.isnan(x)` should be sufficient? np.nan is a float

Reply:
`np.isnan(x)` throws errors when the input argument x is not of the expected type float

(eg - `np.isnan('a')` throws `TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''` )

Hence had added this check for the non np.nan scenarios

Reply:
Gotcha, could you use `isinstance` instead of `type(..) == ...`

Reply:
sure, have updated the PR